File: SUN/read.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import pandas as pd

d = pd.read_excel('hierarchy_three_levels/three_levels.xlsx',skiprows=1)

d_txt = d.\
 melt(id_vars=['category','indoor','outdoor, natural','outdoor, man-made'],var_name='specific').\
 query('value==1').\
 drop(['value']).\
 melt(id_vars=['category','specific'],var_name='generic').\
 query('value==1').\
 filter(['category','specific','generic'])
d_txt.to_csv('levels.txt',sep="\t",doublequote=False,index=F)

File: SUN/read.R
#!/usr/bin/env Rscript
library(readxl)
library(dplyr)
library(tidyr)

d <- read_xlsx('hierarchy_three_levels/three_levels.xlsx',skip=1)

d.txt <-
   d %>% 
   gather('specific','value',-category,-indoor,-`outdoor, natural`,-`outdoor, man-made`) %>%
   filter(value==1) %>% select(-value) %>%
   gather('generic','value',-category,-specific) %>%
   filter(value==1) %>% select(-value) 
write.table(d.txt,'levels.txt',quote=F,row.names=F,sep="\t")
